### ISSUE TITLE
ERR Cannot read property 'contributes' of undefined: TypeError: Cannot read property 'contributes' of undefined

### ISSUE THREAD

**Issue by Vaelek** [Microsoft/azuredatastudio#4252]
- Azure Data Studio Version: 
1.45

Steps to Reproduce:

1. Install any VSIX extension
2. Uninstall that extension

```
 INFO Uninstalling the extension redgate.sql-search-0.3.1 from window 1
D:\Program Files\Azure Data Studio\resources\app\node_modules\zone.js\dist\zone-node.js:2280   ERR Cannot read property 'contributes' of undefined: TypeError: Cannot read property 'contributes' of undefined
    at file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:199416:40
    at CompletePromise_ctor.CompletePromise_then [as then] (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:17896:49)
    at file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:199414:50
    at Emitter.fire (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:20054:38)
    at file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:20179:70
    at Emitter.fire (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:20054:38)
    at Emitter.fire (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:20057:41)
    at ChannelClient.requestEvent.arg.handlers.(anonymous function) (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:30427:70)
    at ChannelClient.onMessage (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:30497:17)
    at file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:30380:89
    at Emitter.fire (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:20054:38)
    at acceptChunk (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:30828:46)
    at Socket.Protocol.Protocol.firstDataChunk._socketDataListener (file:///D:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.main.js:30854:17)
    at ZoneDelegate.invokeTask (D:\Program Files\Azure Data Studio\resources\app\node_modules\zone.js\dist\zone-node.js:421:31)
    at Zone.runTask (D:\Program Files\Azure Data Studio\resources\app\node_modules\zone.js\dist\zone-node.js:188:47)
    at ZoneTask.invokeTask (D:\Program Files\Azure Data Studio\resources\app\node_modules\zone.js\dist\zone-node.js:496:34)
    at Socket.ZoneTask.ZoneTask.cancelFn.invoke (D:\Program Files\Azure Data Studio\resources\app\node_modules\zone.js\dist\zone-node.js:485:48)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at Pipe.onread (net.js:594:20)
D:\Program Files\Azure Data Studio\resources\app\node_modules\zone.js\dist\zone-node.js:2280 WARNING: Promise with no error callback:100
```

I noticed this while developing my first extension. Spent some hours trying to figure out what I did wrong, then realized it did this when uninstalling any extension. In the end, the extension folder is not removed. As was suggested in a Twitter response, uninstalling does not actually remove any files, by design. That makes there effectively be no difference between Disable and Uninstall, except when choosing Uninstall you must go through the steps of getting the vsix again to reinstall even though you still have the files there, that if the version hasn't changed, will simply be re-enabled, and every outdated version you still have will never be removed. 

So I guess this is really 2 issues, one in the error that's shown upon uninstalling anything, and the fact that "Uninstalling" is not an accurate description of what is occurring.

**Comment by kburtram** [comment]
@alanrenmsft this doesn't sound like the expected behavior.  Was there a regression in Jan release?

**Comment by Charles-Gagnon** [comment]
@Vaelek If you uninstall an extension it should delete it if you close and relaunch ADS - is this not happening for you?

**Comment by alanrenmsft** [comment]
@Vaelek I just tried your scenario and I am able to see the same error, but that is not blocking you from uninstalling it. If you uninstall an extension, you will see a file named .obsolete created under the %HOMEPATH%/extensions folder. this file will be there until you restart ADS (note: this is not reload window), after restart the .obsolete file and uninstalled extension folder will both be removed.

**Comment by Vaelek** [comment]
@Charles-Gagnon @alanrenmsft Ok yes, I have been reloading, and any time I did close it, I guess I expected it to be deleted at exit rather than next launch. I did just confirm that at next launch after uninstalling, the folder is in fact removed. the .obsolete file remains, containing previous versions but not the one I just uninstalled, presumably because I had deleted those folders manually through the course of trying to figure this out.
